src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/AuthenticationHelper.NtAuth.cs

```diff
@@ -110,7 +135,7 @@ private static async Task<HttpResponseMessage> SendWithNtAuthAsync(HttpRequestMe
                                 hostName = result.HostName;
                             }
 
-                            if (!isProxyAuth && !authUri.IsDefaultPort)
+                            if (!isProxyAuth && !authUri.IsDefaultPort && UsePortInSpn)
```

Comment:
Shouldn't this be cached rather than hitting the environment variable on every access?

Reply:
I was wondering about that but this is what we did for HttpHandler. I can fix this and do one time lookup and update the tests to use RemoteExec...

Reply:
added `Lazy<bool> s_UsePortInSpn`.